pkg/storage/replica.go

```diff
@@ -3275,6 +3275,13 @@ func (r *Replica) processRaftCommand(
 			}
 			return errors.Errorf("lease %s not held", l)
 		}
+		if raftCmd.BatchRequest.IsSingleSkipLeaseCheckRequest() {
+			// For lease commands, no additional verification is needed if the origin
+			// lease isn't valid.
+			if !r.isLeaseValidLocked(raftCmd.OriginLease, ts) {
```

Comment:
The test failure in `TestRefreshPendingCommands` prompted it. Per @andreimatei's comment, this likely isn't correct, but something bad is happening here. The scenario is that this node (node 3) issued a `RequestLease` command when the previous expired lease was held by node 1. This `RequestLease` command is failing on the `Equivalent` check below because the current lease is held by node 3. Take a look at the log output attached to #12446.